src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.wasm.cs

```diff
@@ -74,24 +70,48 @@ public static void SetBorder(UIElement element, Thickness thickness, Brush brush
 			else
 			{
 				var borderWidth = $"{thickness.Top.ToStringInvariant()}px {thickness.Right.ToStringInvariant()}px {thickness.Bottom.ToStringInvariant()}px {thickness.Left.ToStringInvariant()}px";
+
+				static void ApplySolidColor(UIElement element, Color color, string borderWidth)
+				{
+					element.SetStyle(
+						("border", ""),
+						("border-style", "solid"),
+						("border-color", color.ToHexString()),
+						("border-width", borderWidth));
+				}
+
+				static void ApplyGradient(UIElement element, GradientBrush gradient, string borderWidth)
+				{
+					var border = gradient.ToCssString(element.RenderSize); // TODO: Reevaluate when size is changing
+					element.SetStyle(
+						("border-style", "solid"),
+						("border-color", ""),
+						("border-image", border),
+						("border-width", borderWidth),
```

Comment:
As we're using this a lot, it makes sense to make a specialized invocation for those two SetStyle methods.